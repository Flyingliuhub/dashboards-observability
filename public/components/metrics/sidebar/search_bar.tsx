/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiSearchBar } from '@elastic/eui';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { debounce } from 'lodash';
import { useMetricStore, searchSelector } from '../metrics_store';

export const SearchBar = () => {
  const searchText = useMetricStore(searchSelector);
  const setSearch = useMetricStore((state) => state.setSearch);

  const onChange = debounce(({ query }) => {
    setSearch(query.text);
  }, 300);

  return (
    <div className="metrics-search-bar-input" data-test-subj="metricsSearch">
      <EuiSearchBar
        box={{
          placeholder: 'Search for metrics',
          incremental: true,
        }}
        defaultQuery={''}
        query={searchText}
        onChange={onChange}
      />
    </div>
  );
};
