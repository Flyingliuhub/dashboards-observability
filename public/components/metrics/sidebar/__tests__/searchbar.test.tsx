/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { SearchBar } from '../search_bar';

describe('Search Bar Component', () => {
  configure({ adapter: new Adapter() });

  it('Search Side Bar Component', async () => {
    const wrapper = mount(<SearchBar />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
