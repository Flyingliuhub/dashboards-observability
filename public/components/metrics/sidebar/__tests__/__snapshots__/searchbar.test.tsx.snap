// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component Search Side Bar Component 1`] = `
<Component>
  <div
    className="metrics-search-bar-input"
    data-test-subj="metricsSearch"
  >
    <EuiSearchBar
      box={
        Object {
          "incremental": true,
          "placeholder": "Search for metrics",
        }
      }
      defaultQuery=""
      onChange={[Function]}
      query=""
    >
      <EuiFlexGroup
        alignItems="center"
        gutterSize="m"
        wrap={true}
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
        >
          <EuiFlexItem
            className="euiSearchBar__searchHolder"
            grow={true}
          >
            <div
              className="euiFlexItem euiSearchBar__searchHolder"
            >
              <EuiSearchBox
                incremental={true}
                isInvalid={false}
                onSearch={[Function]}
                placeholder="Search for metrics"
                query=""
              >
                <EuiFieldSearch
                  aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
                  compressed={false}
                  defaultValue=""
                  fullWidth={true}
                  incremental={true}
                  inputRef={[Function]}
                  isClearable={true}
                  isInvalid={false}
                  isLoading={false}
                  onSearch={[Function]}
                  placeholder="Search for metrics"
                >
                  <EuiFormControlLayout
                    compressed={false}
                    fullWidth={true}
                    icon="search"
                    isLoading={false}
                  >
                    <div
                      className="euiFormControlLayout euiFormControlLayout--fullWidth"
                    >
                      <div
                        className="euiFormControlLayout__childrenWrapper"
                      >
                        <EuiValidatableControl
                          isInvalid={false}
                        >
                          <input
                            aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
                            className="euiFieldSearch euiFieldSearch--fullWidth"
                            defaultValue=""
                            onKeyUp={[Function]}
                            placeholder="Search for metrics"
                            type="search"
                          />
                        </EuiValidatableControl>
                        <EuiFormControlLayoutIcons
                          compressed={false}
                          icon="search"
                          isLoading={false}
                        >
                          <div
                            className="euiFormControlLayoutIcons"
                          >
                            <EuiFormControlLayoutCustomIcon
                              size="m"
                              type="search"
                            >
                              <span
                                className="euiFormControlLayoutCustomIcon"
                              >
                                <EuiIcon
                                  aria-hidden="true"
                                  className="euiFormControlLayoutCustomIcon__icon"
                                  size="m"
                                  type="search"
                                >
                                  <EuiIconEmpty
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                    focusable="false"
                                    role="img"
                                    style={null}
                                  >
                                    <svg
                                      aria-hidden={true}
                                      className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                      focusable="false"
                                      height={16}
                                      role="img"
                                      style={null}
                                      viewBox="0 0 16 16"
                                      width={16}
                                      xmlns="http://www.w3.org/2000/svg"
                                    />
                                  </EuiIconEmpty>
                                </EuiIcon>
                              </span>
                            </EuiFormControlLayoutCustomIcon>
                          </div>
                        </EuiFormControlLayoutIcons>
                      </div>
                    </div>
                  </EuiFormControlLayout>
                </EuiFieldSearch>
              </EuiSearchBox>
            </div>
          </EuiFlexItem>
        </div>
      </EuiFlexGroup>
    </EuiSearchBar>
  </div>
</Component>
`;
